### PR TITLE
api: copy ngl_config.hud_export_filename internally

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: tests/api.py [hunk 1]
```diff
@@ -160,6 +162,19 @@ def api_hud(width=234, height=123):
...
+    with tempfile.TemporaryDirectory(prefix="ngl_") as td:
+        fname = op.join(td, 'hud_stats.csv')
+        ctx = ngl.Context()
+        assert ctx.configure(offscreen=1, width=width, height=height, backend=_backend, hud=1, hud_export_filename=fname) == 0
+        del fname
+        scene = _get_scene()
+        assert ctx.set_scene(scene) == 0
+        for i in range(60 * 3):
+            assert ctx.draw(i / 60.) == 0
+        del ctx
```
Comment (cboesch-gpsw) [inline tests/api.py:175]: We already have `cmp_resources.py` which provides a HUD+CSV backed `test_resources` decorator, which we use in `tests/media.py`. Do we really need this test?

File: libnodegl/api.c [hunk 2]
```diff
@@ -95,6 +95,12 @@ static int cmd_stop(struct ngl_ctx *s, void *arg)
     return 0;
 }
 
+static void config_reset(struct ngl_config *config)
+{
+    ngli_freep(&config->hud_export_filename);
+    memset(config, 0, sizeof(*config));
```
Comment (cboesch-gpsw) [inline libnodegl/api.c:101]: I would add the `memset` in a dedicated commit, it resets much more data than before.

**Review by cboesch-gpsw — CHANGES_REQUESTED** [review]

File: tests/api.py (hunk 1 above)
Comment (mbouron-gpsw) [reply tests/api.py:175]: I didn't remember this test. Commit removed.

File: libnodegl/api.c (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/api.c:101]: Done.

**Review by cboesch-gpsw — APPROVED** [review]

**Pull request merged**